libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/PushNotificationManager.swift

```diff
@@ -198,16 +190,147 @@ public class PushNotificationManager: NSObject {
+    ///   - completionBlock: completion block to call with success or failure
+    public func registerForSalesforceNotifications(user: UserAccount, completionBlock:@escaping (Result<Bool, PushNotificationManagerError>)->()) {
+        _ = registerSalesforceNotifications(for: user) {
+            return completionBlock(.success(true))
+        } failBlock: {
+            completionBlock(.failure(.registrationFailed))
+        }
+    }
+    
+    /// Unregister from notifications with Salesforce for a current user
+    /// - Parameter completionBlock: completion block to call with success or failure
+    public func unregisterForSalesforceNotifications(_ completionBlock:@escaping (Bool)->()) {
+        
+        guard let currentUser = UserAccountManager.shared.currentUserAccount else {
+            completionBlock(false)
+            return
+        }
+        self.unregisterForSalesforceNotifications(user: currentUser, completionBlock)
+    }
+    
+    ///  Unregister from Salesforce notfications for a specific user
+    /// - Parameters:
+    ///   - user: The user that should be unregistered from notfications
+    ///   - completionBlock: completion block to call with success or failure
+    public func unregisterForSalesforceNotifications(user: UserAccount, _ completionBlock:@escaping (Bool)->()) {
+        let result = unregisterSalesforceNotifications(for: user) {
+            completionBlock(true)
+        }
+        
+        if (!result) {
+            completionBlock(false)
+        }
+    }
+
+}
+
+private extension PushNotificationManager {
+    
+    private func setupNotificationObservers() {
+        loginObserver =  NotificationCenter.default.addObserver(
+            forName: UserAccountManager.didLogInUser,
+            object: nil,
+            queue: .main
+        ) { [weak self] in self?.onUserLoggedIn($0) }
+        
+        enterForegroundObserver = NotificationCenter.default.addObserver(
+            forName: UIApplication.willEnterForegroundNotification,
+            object: nil,
+            queue: .main
+        ) { [weak self] in self?.onAppWillEnterForeground($0) }
+    }
+    
+    @objc private func onUserLoggedIn(_ notification: Notification) {
+        if deviceToken != nil {
+            SFSDKCoreLogger.i(Self.self, message: "User logged in, registering push")
+            _ = registerSalesforceNotifications(completionBlock: nil, failBlock: nil)
+        }
+    }
+    
+    @objc private func onAppWillEnterForeground(_ notification: Notification) {
+        guard registerOnForeground,
+              !UserAccountManager.shared.isLogoutSettingEnabled,
+              deviceToken != nil else {
+            return
+        }
+        
+        SFSDKCoreLogger.i(Self.self, message: "App entering foreground, re-registering push")
+        _ = registerSalesforceNotifications(completionBlock: nil, failBlock: nil)
+    }
+    
+    private func registerSalesforceNotifications(for user: UserAccount,
```

Comment:
Like unregistering, registering notifications for a given user was a public API in 13.0

Reply:
Yeah, Kept thinking those were typo errors by me.  with `completionBlock` in teh Objc header naming.